src/sinks/tap.rs

```diff
@@ -0,0 +1,170 @@
+use super::util::StreamSink;
+use crate::{
+    buffers::Acker,
+    config::{DataType, SinkContext, SinkOuter},
+};
+use crate::{
+    config::SinkConfig,
+    event::{Event, LogEvent},
+};
+use futures::{future, stream::BoxStream, FutureExt, StreamExt};
+use parking_lot::RwLock;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tokio::sync::broadcast;
+
+type Sender = broadcast::Sender<LogEvent>;
+type Receiver = broadcast::Receiver<LogEvent>;
+
+#[derive(Debug)]
+pub struct TapSink {
+    tx: Sender,
+    acker: Acker,
+}
+
+/// Tap sink is used to 'tap' into events received by uptream components, and broadcast
+/// them to subscribers. Typically, this is used to expose events to the API, but is general
+/// purpose enought that it could technically be used with other mechanisms. This sink is
+/// not added to inventory; it's not intended to be user configurable.
+impl TapSink {
+    pub fn new(tx: Sender, acker: Acker) -> Self {
+        Self { tx, acker }
+    }
+}
+
+#[async_trait::async_trait]
+impl StreamSink for TapSink {
+    async fn run(&mut self, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
+        while let Some(ev) = input.next().await {
+            if let Event::Log(ev) = ev {
+                // This can suffer from TOC/TOU, but the risk is minimal as the purpose
+                // here is solely to reduce expense.
+                if self.tx.receiver_count() > 0 {
+                    let _ = self.tx.send(ev.clone());
```

Comment:
```suggestion
        while let Some(event) = input.next().await {
            if let Event::Log(event) = event {
                // This can suffer from TOC/TOU, but the risk is minimal as the purpose
                // here is solely to reduce expense.
                if self.tx.receiver_count() > 0 {
                    let _ = self.tx.send(event.clone());
```